test/C/src/federated/LoopDistributedDecentralizedPrecedence.lf

```diff
@@ -0,0 +1,85 @@
+/**
+ * This tests that the precedence order of reaction invocation is kept when a feedback loop is
+ * present in decentralized coordination. The Looper reactors have the default zero STP offset, so
+ * they can advance their time to the timer time immediately. Their second reaction, however, has
+ * larger STAA offset (100 ms), so they stall to wait for input. The test ensures that their third
+ * reaction is not invoked before the second.
+ *
+ * To tolerate timing variability (particularly in the CI tests), if the third reaction is invoked
+ * at a tag without the second reaction having been invoked, then it checks that the lag is at least
+ * 100 ms. This means that the input was late by at least 100 ms and was therefore assumed to be
+ * absent, as is normal for decentralized coordination. If the input later arrives, then there will
+ * be an STP violation, which will result in printing a warning, but not in a test failure.
+ *
+ * This test is designed to not be flaky. The timing properties are set so that warnings are printed
+ * if the execution timing is wildly off, specifically if communication between federates takes more
+ * than 100 ms. But the test will not fail in these cases.
```

Comment:
This is great. Having looked at the code for `LoopDistributedDecentralizedPrecedence` and `LoopDistributedDecentralizedPrecedenceHierarchy`, I agree that it should not be possible for them to flake (unless the physical time is non-monotonic, which in light of the known race condition that it has, may be possible in principle -- but that is a separate issue and I am OK with ignoring it considering how rare leap seconds are in practice).